Tweak PR template to be a bit more flexible and lightweight. (#12063)
galaxyproject/galaxy: pull_request_template.md

(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

> ## How to test the changes?

Reviewer: Could there be cases where more than 1 case applies?

Reviewer 2: We could say "at least one", but I never needed more than one.

Reviewer 3: "select all options that apply"? I also had more than one (refactoring, tests exist that cover part of the modified code + I added more tests)

Reviewer 2: Aren't we going back to the version that was confusing to some people?

Reviewer 4: I think the confusion was that the 3rd option could be used for release testing. Writing that in a way that anyone can take this out of context and test it would be a burden for the PR author writing it, and probably not a good way of spending time while testing the release. It should be sufficient for reviewers to understand why the change was made, and IMO can also just point to an issue if it exists.

Saying `select all options that apply` is less ambiguous than picking one, since picking one is tricky for many PRs.

Author: I've used multiple options on occasion also. I like language allowing multiple options.

> (Select one of the following 3 options)

Reviewer 2: ```suggestion
(Select all options that apply)
```

> (Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the change. The "why" may simply be citing a relevant Galaxy issue.) 

Reviewer 5: Now that we are at it, maybe add here a recommendation of including screenshots or gifs for UI components and remove the `For UI Components` section at the end?
If the screenshot is already included ticking the box feels a bit redundant.

> (Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the change. The "why" may simply be citing a relevant Galaxy issue.) 

Reviewer 2: ```suggestion
(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)
```

Author: I removed the fixing a bug language because it felt redundant (on an admittedly spiritual level) with the request to cite an issue - but I'll restore it.

Reviewer 2: Not every bug has an issue, and I don't think we should open one just for the sake of mentioning it in a PR description. Thanks for applying my suggestions!

## How to test the changes? 

> - [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

Reviewer 2: I'd keep the guidance, from the discussion at today's Dev Meeting it seemed it was not clear the author was supposed to choose one option.

```suggestion
(Select one of the following 3 options)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
```

Author: But it could be more than one of these three options right - I've checked multiple ones before? I'll redo it with this language though. I guess the important change is to make it clear if there first two are checked we don't need the third.

- [ ] This is a refactoring of components with existing test coverage.

> - [ ] Instructions for manual testing are as follows (only required if no automated tests): 

Reviewer 3: The text in parentheses seems redundant given the line below
